packages/rollup-plugin-html/rollup-plugin-html.js

```diff
@@ -31,26 +32,24 @@ const defaultFileName = 'index.html';
 function rollupPluginHtml(pluginOptions) {
   pluginOptions = {
     inject: true,
+    flatten: true,
     minify: !watchMode,
+    rootDir: process.cwd(),
     ...(pluginOptions || {}),
   };
 
-  /** @type {string} */
-  let inputHtml;
-  /** @type {string[]} */
-  let inputModuleIds;
-  /** @type {Map<string, string>} */
-  let inlineModules;
-  /** @type {string} */
-  let htmlFileName;
+  pluginOptions.rootDir = path.resolve(pluginOptions.rootDir);
+
   /** @type {GeneratedBundle[]} */
-  let generatedBundles;
+  let thisGeneratedBundles;
   /** @type {TransformFunction[]} */
-  let externalTransformFns = [];
+  let thisExternalTransformFns = [];
+  /** @type {HtmlFile[]}  */
+  const thisHtmlFiles = [];
 
   // variables for multi build
   /** @type {string[]} */
-  const multiOutputNames = [];
+  const thisMultiOutputNames = [];
```

Comment:
I don't like the `this` but you probably did this on purpose 😆

Reply:
yeah I had a tough time to understand which variables are local function variabels and which are "plugin" variables...  `this...` seemed better then `global...` 😬

of course open for better suggestions as long as there is distinction

Reply:
For variables distinct from function parameter, I don't see the need. 

For similar we can do something like "all" or "gathered". If the plugin is becoming too large, we can also place more logic in separate functions. `createHtmlAsset` is a good candidate for that, and the child plugin logic.